### PR TITLE
feat(techradar-plugin) add techradar blip history backend support

### PR DESCRIPTION
**Pull request by o-farooq**
## Hey, I just made a Pull Request!

Currently tech radar shows the current state with the blip name, It does not have any history or description which is really important for evolving tech radar . e.g. [ThoughtWorks radar](https://www.thoughtworks.com/radar/techniques?blipid=777) when clicked on blip one can see the whole history and why a blip is in a particular ring

This PR is just to add some data structure to support history, there are no UI changes . description, moved state and ring are included in history instead of root entry. Blip is displayed in latest history entry ring. 

If structure looks good, can work on the UI to display history

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/spotify/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

File: plugins/tech-radar/src/api.ts [hunk 1]
```diff
@@ -34,11 +34,17 @@ export interface RadarQuadrant {
 export interface RadarEntry {
   key: string; // react key
   id: string;
-  moved: MovedState;
-  quadrant: RadarQuadrant;
-  ring: RadarRing;
+  quadrant: string;
   title: string;
   url: string;
+  history: Array<RadarEntryHistory>;
```
Comment (Rugvip) [inline plugins/tech-radar/src/api.ts:40]: I'm not quite getting the feeling that this conveys that `history[0]` is the current state. Wonder if something like calling it `timeline` helps at all? Another possibility would be to go with

```ts
export interface RadarEntrySnapshot {
  date: Date;
  ringId: string;
  description?: string;
  moved?: MovedState;
}

export interface RadarEntry {
  key: string;
  id: string;
  quadrant: string;
  title: string;
  url: string;
  current: RadarEntrySnapshot;
  history?: Array<RadarEntrySnapshot>;
}
```

File: plugins/tech-radar/src/api.ts [hunk 2]
```diff
@@ -34,11 +34,17 @@ export interface RadarQuadrant {
...
+}
+
+export interface RadarEntryHistory {
+  date: Date;
+  ring: string;
```
Comment (Rugvip) [inline plugins/tech-radar/src/api.ts:45]: Thanks you for this normalization, previous types where a bit weird 😅 

Think we should call it `ringId` instead?

**Review by Rugvip — COMMENTED** [review]
Thanks! Couple of ideas around the data structures. Could you also please add a [changeset](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets)? Seeing as we'll want the `RadarRing -> string` fix in regardless, this is gonna be a breaking change, meaning a minor bump as we're on v0

File: plugins/tech-radar/src/api.ts (hunk 1 above)
Comment (o-farooq) [reply plugins/tech-radar/src/api.ts]: renamed to `timeline`, let me know if you want me to add a `current` field

File: plugins/tech-radar/src/api.ts (hunk 2 above)
Comment (o-farooq) [reply plugins/tech-radar/src/api.ts]: renamed to `ringId`

File: plugins/tech-radar/src/components/RadarComponent.tsx [hunk 3]
```diff
@@ -47,6 +48,29 @@ const useTechRadarLoader = (props: TechRadarComponentProps) => {
 const RadarComponent = (props: TechRadarComponentProps): JSX.Element => {
   const { loading, error, value: data } = useTechRadarLoader(props);
 
+  const mapToEntries = (
+    loaderResponse: TechRadarLoaderResponse | undefined,
+  ): Array<Entry> => {
+    return loaderResponse!.entries.map(entry => {
```
Comment (freben) [inline plugins/tech-radar/src/components/RadarComponent.tsx:54]: This one accepts `undefined` and then immediately does `!`, which sounds risky. Did you mean to do `?.` and return the empty array as a fallback or something like that?

**Review by freben — APPROVED** [review]
One question, but will merge for now anyway - if it is something that you think should be addressed, it can be done in a separate PR

**Pull request merged**